src/components/HyvaksymisEsitys/hyvaksymisEsitysFormUtil.ts

```diff
@@ -55,6 +72,21 @@ export default function getDefaultValuesForForm(hyvaksymisEsityksenTiedot: Hyvak
   };
 }
 
+export function transformHyvaksymisEsitysFormToTallennaHyvaksymisEsitysInput(formData: HyvaksymisEsitysForm) {
+  const muistutukset = formData.muokattavaHyvaksymisEsitys.muistutukset;
+  return {
+    ...formData,
+    muokattavaHyvaksymisEsitys: {
+      ...formData.muokattavaHyvaksymisEsitys,
+      muistutukset: muistutukset
+        ? Object.keys(muistutukset).reduce((acc, key) => {
+            return acc.concat(muistutukset[key] ?? []);
+          }, [] as KunnallinenLadattuTiedostoInput[])
+        : [],
```

Comment:
oisko tää parempi

muistutukset: muistutukset ? Object.values(muistutukset).filter((m): m is KunnallinenLadattuTiedostoInput[] => !!m).flat() : null,